### PR TITLE
Third milestone: Game Logic

### PR DESCRIPTION
**Pull request by sergiomauz**
1. Created game.rb file and Game Class:
- Added constructor.
- Added winning combinations.
- Added **valid_movement?** method.
- Added **finished?** method.
- Added **player_win?** method.
- Added **new_movement** method.
- Added **reset** method.
2. Created board.rb file and Board Class.
- Added **print** method.
3. Created player.rb file and Player Class:
- Added constructor.

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,60 +1,9 @@
...
 
-  if first_player_movements.length + second_player_movements.length == 9
-    0
-  else
-    -1
-  end
-end
+require './lib/player.rb'
+require './lib/board.rb'
+require './lib/game.rb'
```
Comment (mcrd25) [inline bin/main.rb]: [OPTIONAL]
Consider using `require_relative` for files in the project.
Require should be used for external files, like gems, while require_relative should be used for referring to files within your directory.

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# rubocop: disable Metrics/CyclomaticComplexity
+class Game
+  attr_reader :switch_player
+  attr_reader :who_won
+
+  WINNING_COMBINATIONS = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]].freeze
```
Comment (mcrd25) [inline lib/game.rb:6]: [OPTIONAL]
Is there a better way to get all winning combinations than hard coding it like this?

**Review by mcrd25 — APPROVED** [review]
# Changes Approved 💪 ⏭ 

Hey @sergiomauz! Hey @peterrobert! 👋 

You did a great job and have met all the requirements for this milestone as far as. I can tell.

Consider my optional comments!
Otherwise, good job! 👏 

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

File: bin/main.rb (hunk 1 above)
Comment (sergiomauz) [reply bin/main.rb]: Done!. Thank you.

File: lib/game.rb (hunk 2 above)
Comment (sergiomauz) [reply lib/game.rb:6]: Considering the logic of this game, using slots with an assigned number, the easier way to work is using hard code, even the number of combinations is only 8.

**Pull request merged**